cmake/Modules/SetGPUArchs.cmake

```diff
@@ -0,0 +1,50 @@
+# =============================================================================
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+set(GPU_ARCHS
+    ""
+    CACHE
+      STRING
+      "List of GPU architectures (semicolon-separated) to be compiled for. Empty string means to auto-detect the GPUs on the current system"
+)
```

Comment:
In a follow up PR we should remove this option and just use the standard `CMAKE_CUDA_ARCHITECTURES` variable

Reply:
Ended up raising an issue about this because overriding the default for `CMAKE_CUDA_ARCHITECTURES` is problematic: https://gitlab.kitware.com/cmake/cmake/-/issues/21302